test/unit/org/apache/cassandra/index/sai/cql/types/collections/lists/FrozenListAsciiTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.cql.types.collections.lists;
+
+import java.util.Collection;
+
+import org.junit.runners.Parameterized;
+
+import org.apache.cassandra.index.sai.cql.types.DataSet;
+import org.apache.cassandra.index.sai.cql.types.IndexingTypeSupport;
+import org.apache.cassandra.index.sai.cql.types.collections.CollectionDataSet;
+
+public class FrozenListAsciiTest extends IndexingTypeSupport
+{
+    @Parameterized.Parameters(name = "dataset={0},wide={1},scenario={2}")
+    public static Collection<Object[]> generateParameters()
+    {
+        return generateParameters(new CollectionDataSet.FrozenListDataSet<>(new DataSet.AsciiDataSet()));
```

Comment:
We have a huge number of test classes, which makes it a little bit tedious to, say, run all the scenarios in the `types.collections.lists` package. What if we condensed down to one class per package (which would then likely remove the need for the packages in the first place)? For example, you could begin by doing something like...

```
public class ListIndexingTest extends IndexingTypeSupport
{
    @Parameterized.Parameters(name = "dataset={0},wide={1},scenario={2}")
    public static Collection<Object[]> generateParameters()
    {
        CollectionDataSet.FrozenListDataSet<String> frozenStringLists = new CollectionDataSet.FrozenListDataSet<>(new DataSet.AsciiDataSet());
        CollectionDataSet.FrozenListDataSet<BigDecimal> frozenDecimalLists = new CollectionDataSet.FrozenListDataSet<>(new DataSet.DecimalDataSet());
        ...

        return Arrays.asList(new Object[][]
                {
                        {frozenStringLists, true, Scenario.MEMTABLE_QUERY},
                        {frozenStringLists, false, Scenario.MEMTABLE_QUERY},
                        {frozenDecimalLists, true, Scenario.MEMTABLE_QUERY},
                        {frozenDecimalLists, false, Scenario.MEMTABLE_QUERY},
                        ...
                });
    }
}
```

Reply:
Same idea for maps, sets, and scalars

Reply:
I've tried this and like the result. The one concern I have about this is that later we are going to end up with 4 more scenarios. This will, potentially, push the run time of each of these tests to ~5 mins. 

Is that going to be an issue?

If it was then we could potentially have a test for each scenario. That would still significantly reduce the number of tests.

Reply:
> My other concern about these is the length of time a combined test will take to run when we add the additional scenarios for the on-disk indexes.

Fair. I suppose we can come back to this.